test/functional/collection_deserialization_tests.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+var test = require('./shared').assert;
```

Comment:
Thank you for writing tests!

1. Since we are testing against Node >= 4, please use ES6 syntax where possible. That includes
    - `const` where possible, `let` otherwise.
    - `class` instead of raw functions and prototypes.
    - template strings (``` ` ` ```) instead of string concatenation

2. instead of using our old assert library (which we mostly keep around for legacy reason), please use `chai.expect` for your assertions.